src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -168,9 +216,20 @@ bool TryRemoveRange(ref TableValueReader reader, out DateTime next)
                         if (baseline > end)
                             return false;
 
+                        // we need to get the next segment before the actual remove, since it might lead to a split
+                        next = TryGetNextSegment(baseline);
+
                         var newSegment = new TimeSeriesValuesSegment(holder.Allocator.Buffer.Ptr, MaxSegmentSize);
                         newSegment.Initialize(readOnlySegment.NumberOfValues);
 
+                        if (baseline >= from && end <= to)
```

Comment:
How would this work with replication?
And what if there is a conflict?

Reply:
We overwrite the existing segment with a new one.
So it should be replicated and appended as an entire segment. 
If there is a conflict it will be merged with the other segment and replicated back and replace the pending deletion segment (whether is was already deleted or not).